### PR TITLE
Bytes refactor

### PR DESCRIPTION
**Pull request by fi-ya**
In this pull request I have refactored `ClientHandler,processSendResponse` to send to convert the response string into a bytes and write to `clientSocket.getOutputStream`. I have added byte attributes and setter methods to `Response` class. 

I will now be able to process files in the next PR. As i will be able to convert the file to string and create a Response object as normal and then convert the string to bytes which i will send to the client. 

🧪 Codecov coverage: 80%
❓ Gradle test coverage: Class:93% Method:87% Line:80%

### CODE REVIEW

**Review by andy8thlight — CHANGES_REQUESTED** [review]
A coupe of things I'm not sure of and there maybe a fairly big flaw in here around exception handling.

File: src/test/java/org/httpserver/client/ClientHandlerTest.java [hunk 1]
```diff
@@ -4,25 +4,24 @@
...
-        Response mockResponse = mock(Response.class);
-        when(mockResponse.stringFormatResponse()).thenReturn(mockResponseString);
 
-        clientHandler.sendResponse(mockResponse, mockClientResponseWriter);
+        byte[] actualByte = clientHandler.responseStringToBytes(mockResponse);
 
-        verify(mockClientResponseWriter).write("HTTP/1.1 200 OK");
-        verify(mockClientResponseWriter, times(1)).close();
+        byte[] expectedByte = {72, 84, 84, 80, 47, 49, 46, 49, 32, 50, 48, 48, 32, 79, 75};
+        assertEquals(Arrays.toString(expectedByte), Arrays.toString(expectedByte));
```
[the remark this answers is not in the document]
Comment (fi-ya) [reply src/test/java/org/httpserver/client/ClientHandlerTest.java:24]: Yes, i can refactor to `"HTTP/1.1 200 OK".getBytes();`

File: src/main/java/org/httpserver/client/ClientHandler.java [hunk 2]
```diff
@@ -29,20 +29,26 @@ public void updateClientConnectionLogger() {
...
 
-    public void sendResponse(Response response, PrintWriter clientResponseWriter) {
-        clientResponseWriter.write(response.stringFormatResponse());
-        clientResponseWriter.close();
+    public void sendResponse(byte[] responseByte) throws IOException {
+        OutputStream clientOutputStream = clientSocket.getOutputStream();
+        clientOutputStream.write(responseByte);
+        clientOutputStream.flush();
+        clientOutputStream.close();
     }
```
[the remark this answers is not in the document]
Comment (fi-ya) [reply src/main/java/org/httpserver/client/ClientHandler.java:52]: Thanks, ive added a try with resource statement

**Review by devlinmorrow — APPROVED** [review]
So the idea is to have the Response object contain both the String versions and the bytes versions? Yes I think that seems like it's one logical way of approaching this. I can imagine there could be some more robust testing of different things that can happen in a Response, and different ways it could fail, but for now I think this is good.

**Pull request merged**